Add Packer template for OpenBSD (#1)
devinsolutions/openbsd-packer-template: README.md

# OpenBSD Packer template

## Goal

> Goal of this project is to provide a small OpenBSD Vagrant and GCP images, primarily for testing

Reviewer: ```suggestion
Goal of this project is to provide compact OpenBSD Vagrant and GCE images, primarily for testing
```

Ansible roles.

Vagrant Box is available here: https://app.vagrantup.com/devinsolutions/boxes/openbsd-66-basic

By "basic" we mean that only `bsd`, `bsd.rd`, `base` and `comp` sets are included (plus GCE daemons
and Python needed by them in GCE images). We support only latest stable OpenBSD release. No binary

> patches are included.

Reviewer: Why are binary patches not being included?

Author: I don't want to have half-patched images, therefore I see only two options: either include no binary patches, or update these images as soon as those patches become available. Since I have no automation for this set up right now and I don't want to update these images manually every time, I've decided not to include these patches.

Primary purpose is testing Ansible roles anyway, and one can install these patches before the test begins (e.g. in Molecule prepare playbook) if needed.

Reviewer: I think that unpatched images are worse than half-patched. How about we provide up-to-date images on best-effort basis and share the responsibility for updating them among the whole admin team? Combined with installing patches before running the tests, we should be always secure while achieving the best performance to cost ratio.

Author: That sounds good. I've added a provisioner which installs all available binary patches.

## YAML

The template is written in YAML, not in JSON (we want comments and anchors to keep the template
DRY). We use [yacker](https://yacker.readthedocs.io/en/latest/) to invoke Packer.

## Requirements

  - POSIX shell and standard utilities
  - Pipenv (or your preferred way of installing Python dependencies, that are listed in `Pipfile`)
  - Packer
  - VirtualBox
  - `qemu-img` (if building image for GCE)
  - `jq` (if building image for GCE)

## Building

  - Install Python dependencies (`$ pipenv sync`).

>   - Decide whether you want to build GCE image of Vagrant Box. You can specify them using `-only`

Reviewer: ```suggestion
  - Decide whether you want to build GCE image or Vagrant Box. You can specify them using `-only`
```

Or did you wanna say something else?

    flag. See the following sections for prerequisites specific for each builder.
  - Copy `vars.yaml.template` to `vars.yaml` and fill in sections you are interested in.
  - `$ mkdir -p output`
  - `$ pipenv run yacker build -var-file=vars.yaml [-only=vbox-gce-builder,vbox-vagrant-builder]
     openbsd.yaml`

During build, you may see several errors like `==> vbox-gce-builder: ksh: sudo: not found`. These
are expected and it is safe to ignore them.

### Building and uploading GCE image (`vbox-gce-builder`)

Read and understand how does
[`googlecompute-import`](https://www.packer.io/docs/post-processors/googlecompute-import.html)
post-processor work. Fill-in required info to `vars.yaml`. The resulting image will be named

> `openbsd-66-minimal`, make sure that image with this name **does not** exist in the target project,

Reviewer: ```suggestion
`openbsd-66-basic`, make sure that image with this name **does not** exist in the target project,
```

For consistency.
